ddtrace/propagation/http.py

```diff
@@ -550,35 +554,29 @@ def _get_traceparent_values(tp):
         Otherwise we extract the trace-id, span-id, and sampling priority from the
         traceparent header.
         """
+        tp = tp.strip()
+        if not re.match(_TRACEPARENT_HEX_REGEX, tp) or tp.startswith("ff"):
```

Comment:
What's the benefit of adding this regex?

Reply:
The regex allows us to combine multiple checks into one. This regex validates that the traceparent does not have uppercase letters and that all the components of the traceparent have the correct length.

Reply:
It's an optimization to the bug fixes that were introduced earlier.